### PR TITLE
Delivered feedback mechanism

### PR DESCRIPTION
**Pull request by ThomasPoot**
Mass-reformatting and standardization of names
Delivered feedback mechanism added
Added further queue access
Added testing docker-compose file
Naming fix to a template file (ADDRESS_TYPE_CHANGED_CEEST)
Sorted the outcome code list

### CODE REVIEW

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/service/impl/RmFieldRepublishService.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.ons.census.fwmt.outcomeservice.service.impl;
```
Comment (spdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/outcomeservice/service/impl/RmFieldRepublishService.java:1]: Move this to 
uk.gov.ons.census.fwmt.outcomeservice.message
and 
rename it to match the other Classes in this file RmFieldRepublishProducer (consistency)

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgAddressNotValidProcessor.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  @Autowired
+  private SpgReasonCodeLookup spgReasonCodeLookup;
+
+  @Override
+  public UUID process(SpgOutcomeSuperSetDto outcome, UUID caseIdHolder) throws GatewayException {
+    UUID caseId = (outcome.getCaseId() != null) ? outcome.getCaseId() : caseIdHolder;
+
+    String reasonCode = spgReasonCodeLookup.getLookup(outcome.getOutcomeCode());
+
+    String eventDateTime = outcome.getEventDate().toString();
```
Comment (spdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgAddressNotValidProcessor.java:40]: Use SimpleDateFormatter to convert to correct format YYYY-MM-DD THH:MM:SS.ssssssZ

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgAddressTypeChangedCeEstProcessor.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  @Autowired
+  private GatewayCacheService gatewayCacheService;
+
+  @Override
+  public UUID process(SpgOutcomeSuperSetDto outcome, UUID caseIdHolder) throws GatewayException {
+    UUID caseId = (outcome.getCaseId() != null) ? outcome.getCaseId() : caseIdHolder;
+    cacheData(outcome, caseId);
+
+    Map<String, Object> root = new HashMap<>();
+    String eventDateTime = outcome.getEventDate().toString();
```
Comment (spdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgAddressTypeChangedCeEstProcessor.java:42]: Use SimpleDateFormatter to convert to correct format YYYY-MM-DD THH:MM:SS.ssssssZ

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgAddressTypeChangedHhProcessor.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  @Autowired
+  private GatewayCacheService gatewayCacheService;
+
+  @Override
+  public UUID process(SpgOutcomeSuperSetDto outcome, UUID caseIdHolder) throws GatewayException {
+    UUID caseId = (outcome.getCaseId() != null) ? outcome.getCaseId() : caseIdHolder;
+    cacheData(outcome, caseId);
+
+    Map<String, Object> root = new HashMap<>();
+    String eventDateTime = outcome.getEventDate().toString();
```
Comment (spdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgAddressTypeChangedHhProcessor.java:42]: Use SimpleDateFormatter to convert to correct format YYYY-MM-DD THH:MM:SS.ssssssZ

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgExtraordinaryRefusalReceivedProcessor.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  @Autowired
+  private GatewayOutcomeProducer gatewayOutcomeProducer;
+
+  @Autowired
+  private GatewayEventManager gatewayEventManager;
+
+  @Override
+  public UUID process(SpgOutcomeSuperSetDto outcome, UUID caseIdHolder) throws GatewayException {
+    UUID caseId = (outcome.getCaseId() != null) ? outcome.getCaseId() : caseIdHolder;
+    String eventDateTime = outcome.getEventDate().toString();
```
Comment (spdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgExtraordinaryRefusalReceivedProcessor.java:33]: Use SimpleDateFormatter to convert to correct format YYYY-MM-DD THH:MM:SS.ssssssZ

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgFulfilmentRequestProcessor.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+  @Override
+  public UUID process(SpgOutcomeSuperSetDto outcome, UUID caseIdHolder) throws GatewayException {
+    UUID caseId = (outcome.getCaseId() != null) ? outcome.getCaseId() : caseIdHolder;
+    if (outcome.getFulfillmentRequests() == null) {
+      return caseIdHolder;
+    }
+    for (FulfilmentRequestDto fulfilmentRequest : outcome.getFulfillmentRequests()) {
+      if (!isQuestionnaireLinked(fulfilmentRequest)) {
+
+        String eventDateTime = outcome.getEventDate().toString();
```
Comment (spdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgFulfilmentRequestProcessor.java:60]: Use SimpleDateFormatter to convert to correct format YYYY-MM-DD THH:MM:SS.ssssssZ

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgHardRefusalReceivedProcessor.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  @Autowired
+  private GatewayOutcomeProducer gatewayOutcomeProducer;
+
+  @Autowired
+  private GatewayEventManager gatewayEventManager;
+
+  @Override
+  public UUID process(SpgOutcomeSuperSetDto outcome, UUID caseIdHolder) throws GatewayException {
+    UUID caseId = (outcome.getCaseId() != null) ? outcome.getCaseId() : caseIdHolder;
+    String eventDateTime = outcome.getEventDate().toString();
```
Comment (spdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgHardRefusalReceivedProcessor.java:34]: Use SimpleDateFormatter to convert to correct format YYYY-MM-DD THH:MM:SS.ssssssZ

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgLinkedQidProcessor.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  @Override
+  public UUID process(SpgOutcomeSuperSetDto outcome, UUID caseIdHolder) throws GatewayException {
+    if (outcome.getFulfillmentRequests() == null) {
+      return caseIdHolder;
+    }
+    for (FulfilmentRequestDto fulfilmentRequest : outcome.getFulfillmentRequests()) {
+      if (isQuestionnaireLinked(fulfilmentRequest)) {
+        String eventDateTime = outcome.getEventDate().toString();
```
Comment (spdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgLinkedQidProcessor.java:49]: Use SimpleDateFormatter to convert to correct format YYYY-MM-DD THH:MM:SS.ssssssZ

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/service/impl/RmFieldRepublishService.java (hunk 1 above)
Comment (ThomasPoot) [reply src/main/java/uk/gov/ons/census/fwmt/outcomeservice/service/impl/RmFieldRepublishService.java]: Done.

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgAddressNotValidProcessor.java (hunk 2 above)
Comment (ThomasPoot) [reply src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgAddressNotValidProcessor.java]: Done.

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgAddressTypeChangedCeEstProcessor.java (hunk 3 above)
Comment (ThomasPoot) [reply src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgAddressTypeChangedCeEstProcessor.java]: Done.

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgAddressTypeChangedHhProcessor.java (hunk 4 above)
Comment (ThomasPoot) [reply src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgAddressTypeChangedHhProcessor.java]: Done.

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgExtraordinaryRefusalReceivedProcessor.java (hunk 5 above)
Comment (ThomasPoot) [reply src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgExtraordinaryRefusalReceivedProcessor.java]: Done.

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgFulfilmentRequestProcessor.java (hunk 6 above)
Comment (ThomasPoot) [reply src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgFulfilmentRequestProcessor.java]: Done.

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgHardRefusalReceivedProcessor.java (hunk 7 above)
Comment (ThomasPoot) [reply src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgHardRefusalReceivedProcessor.java]: Done.

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgLinkedQidProcessor.java (hunk 8 above)
Comment (ThomasPoot) [reply src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgLinkedQidProcessor.java]: Done.

**Pull request merged**